src/main/java/pl/wrapper/parking/result/Result.java

```diff
@@ -0,0 +1,52 @@
+package pl.wrapper.parking.result;
+
+import lombok.Getter;
+import lombok.Setter;
+
+/**
+ * Class {@code Result} is a wrapper for any returning value and
+ * can storage any exception as its type and error description.
+ *<p>
+ * {@code Result} is containing 2 static methods for generating
+ * a correct {@code Result} instance:
+ * <ul>
+ * <li>{@code Success}: generate the success instance</li>
+ * <li>{@code Failure}: generate the failure instance</li>
+ * </ul>
+ *
+ * @param <T> the type of the result
+ */
+@Getter
+@Setter
+public class Result<T> {
```

Comment:
use derived classes from the Result base class with overridden methods instead of the 'success' field.